codemagic.yaml

```diff
@@ -93,31 +139,25 @@ definitions:
       name: Test Android
       script: |
         cd project
-        ./gradlew connectedAndroidTest
-        result=$?
-        adb logcat -d > emulator.log
-        if [ $result -eq 0 ]
-        then
-          echo "Tests executed successfully"
-        else
-          echo "Tests failed"
-          exit 1
-        fi
+        sh ../ci/run_tests.sh
     # Run library sample app tests
     - &stepLibAppRunTests
       name: Test Android
       script: |
         cd projectlib
-        ./gradlew connectedAndroidTest
-        result=$?
-        adb logcat -d > emulator.log
-        if [ $result -eq 0 ]
-        then
-          echo "Tests executed successfully"
-        else
-          echo "Tests failed"
-          exit 1
-        fi
+        sh ../ci/run_tests.sh
+    # Run generated app tests
+    - &stepGeneratedAppRunTests
+      name: Test Android
+      script: |
+        cd p42app
+        sh ../ci/run_tests.sh
+    # Run generated lib sample app tests
+    - &stepGeneratedLibRunTests
+      name: Test Android
+      script: |
+        cd p42lib
+        sh ../ci/run_tests.sh
 
 # Flows
 workflows:
```

Comment:
Just a though in hindsight  maybe makes sense to rename the "develop-app-builds" / "Dev app builds" to something like
"develop-template-builds" / "Dev template builds". Here and in other targets that build the template.
Then it is easier to understand that these build the template and the others build the generated all.

Just a thought that came to mind when looking at the build dialog in Codemagic